scripts/gen-go-sdk.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# This script regenerates the Go SDK for a given HCP service ($1). Pass 'cloud-shared' to regenerate shared models.
+
+# The steps are:
+# 1. Remove the original SDK files for the service if they exist.
+# 2. Fetch the latest public API specs for the service from the central spec repo.
+# 3. Run temporary transformations on those specs to prepare them for SDK generation.
+# 4. Iterate over each stage and version of the service specs and generate the corresponding SDK. 
+#    (Note: Currently only the 'preview' stage is supported)
+# 5. Remove temporary directories.
+
+BOLD='\033[1m'
+GREEN='\033[32m'
+NA='\033[0m' # no attributes (color or format)
+
+generate_sdk() {
+  service=$1
+  stage=$2
+  version=$3
+
+  echo -e "Creating target SDK directory: ${BOLD}hcp-sdk-go/clients/$service/$stage/$version${NA}"
+  mkdir -p "$GOPATH/src/github.com/hashicorp/hcp-sdk-go/clients/$service/$stage/$version"
+
+  echo -e "Generating SDK for ${BOLD}$service${NA} (stage: ${BOLD}$stage${NA}, version: ${BOLD}$version${NA})"
+  swagger generate client \
+    -f "$GOPATH/src/github.com/hashicorp/cloud-api/specs/$service/$stage/$version"/*.swagger.json \
```

Comment:
I made similar comments on the `cloud-api` changes, so sorry for being a broken record on this, but I would really consider writing these scripts in a way that does not rely on absolute paths or sibling checkouts (which I think can be done here, except maybe for hcloud artifacts). This would make the script generally more resilient to being checked out in different directories, and CI environments in particular might have trouble dealing with the absolute paths you're using.

Reply:
Ah yes, will do! Got a little bogged down in the refactor stage, so I left a lot of the original paths.